### ISSUE TITLE
Ability to run a sql script on a db context with "dotnet ef"

### ISSUE THREAD

**Issue by GaijinFizz** [dotnet/efcore#33240]
We are currently running SQL scripts on our database after the migration is run.
The initial migration creates tables but we also need to add stored procedures and events (the mariadb equivalent of scheduled jobs) to that database. There is also a sql script to add mock data which we run once the tables are created.

I understand that we could have a myriad of migrations, one for every new feature for instance, and the mock data could be added via EF. Sometimes it's just more practical to do things in SQL. 
Also, since the project is still in its early days, we just delete the initial migration and recreate it to keep things clean. Once the first version is out there it would be a different story.

Simply put, from "dotnet ef" tool, it would be nice to have the ability to run an SQL script (not an EF migration) directly on a db context. Since we use MariaDB and that the Pomelo package is hard to use in PowerShell, we had to install ODBC drivers and run the SQL scripts that way.

**Comment by ajcvickers** [comment]
@GaijinFizz Note that if you want to get a SQL script for the current state of the `DbContext` (essentially what `EnsureCreated` uses) then `context.Database.GenerateCreateScript();` will do this. Should be easy to wrap a tool around that if you want.

**Comment by GaijinFizz** [comment]
When MariaDB was on the dev machine we were using the tool you mentioned. But we moved it to a docker container, which means we didnt have the tool locally anymore. This is why I have been looking at other ways to run the scripts. 

Currently we have a single Powershell command to
- drop the DB
- remove previous migrations
- create a new migration
- apply the migration
- run the SQL scripts via ODBC

But I didn't think of injecting the commands to run the scripts within the generated migration via a script...

The script could become
- drop the DB
- remove previous migrations
- create a new migration
- inject code to run scripts into the .cs file of the migration
- apply the migration

Thanks for giving me new ideas.

**Comment by roji** [comment]
> But we moved it to a docker container, which means we didnt have the tool locally anymore. This is why I have been looking at other ways to run the scripts.

I'm no expert, but I haven't heard of this as a reason to use programmatic means (ODBC, EF...) just to run arbitrary SQL scripts... I'm still not sure why you can't have the mariadb command-line tool installed wherever it is that you need to run your script (inside or outside the container) - that certainly seems to be the standard practice...

In any case, injecting the SQL into the migration is indeed also an option - though that seems like a somewhat convoluted way to go about things... I'll go ahead and close this as I haven't seen a good reason to introduce EF arbitrary SQL exectuion via direct CLI, but feel free to post back here with further questions/thoughts.

**Comment by GaijinFizz** [comment]
You're right, I just wanted to avoid installing an extra tool when setting up the dev environment.
I could indeed put the scripts in the container and use the command line tool already there.
Thank you.
